sources/osgShader/Compiler.js

```diff
@@ -317,10 +317,12 @@ define( [
             var exist = this._variables[ nameID ];
             if ( exist ) {
                 // see comment in Variable function
-                if ( exist.getType() !== type ) {
+                if ( exist.getType() === type ) {
+                    return exist;
+                }
+                if ( exist.getType() === 'sampler2D' && type !== 'sampler2D' ) {
                     Notify.error( 'Same uniform, but different type' );
                 }
-                return exist;
```

Comment:
error is breaking and the user will get errors on compilation if the node is really used.
(in our case the node isn't used in compilation...)